src/main/scala/com/github/awant/habrareader/actors/LibraryActor.scala

```diff
@@ -47,27 +51,53 @@ class LibraryActor(subscriptionReplyInterval: FiniteDuration, chatData: models.C
       chatData.updateSubscription(chatId, subscribe).onComplete {
         case Success(_) =>
         case Failure(err) => println(err)
-    }
+      }
     case SettingsChanging(chatId: Long, cmd: String) =>
-      val settingsCmd = SettingsRequestParser.parse(cmd)
-      settingsCmd.cmd match {
-        case ChangeCommand.UNKNOWN => Reply(chatId, "Unknown command")
-        case ChangeCommand.RESET => chatData.updateChat(Chat.withDefaultSettings(chatId))
-        case ChangeCommand.CLEAR => chatData.updateChat(Chat.withEmptySettings(chatId))
-        case ChangeCommand.SET => chatData.appendSettingToChat(chatId, settingsCmd.args.head, settingsCmd.args(1))
+      println(s"SettingsChanging($chatId, $cmd)")
+
+      cmd match {
+        case Command("/reset") =>
+          chatData.replaceChat(Chat.withDefaultSettings(chatId))
+        case CommandStringDouble("/author", name, weight) =>
+          chatData.updateChat(chatId)(chat => chat.copy(authorWeights = chat.authorWeights.updated(name, weight)))
+        case CommandStringDouble("/tag", name, weight) =>
+          chatData.updateChat(chatId)(chat => chat.copy(tagWeights = chat.tagWeights.updated(name, weight)))
+        case CommandDouble("/rating", ratingThreshold) =>
+          chatData.updateChat(chatId)(_.copy(ratingThreshold = ratingThreshold))
       }
 
     case SettingsGetting(chatId) =>
       chatData.getChatSettings(chatId).onComplete {
         case Success(settings) => subscribedBot ! Reply(chatId, settings)
-        case Failure(_) => subscribedBot ! Reply(chatId, "")
+        case Failure(err) => subscribedBot ! Reply(chatId, "unknown error")
       }
     case NewPostsSending =>
-      chatData.getUpdates(lastUpdateDate).onComplete {
-        case Success(updates) => updates.foreach{case (chat, post) => subscribedBot ! PostReply(chat.id, post)}
+      val currentLast = chatDataLastTime
+
+      chatData.getUpdates(currentLast).onComplete {
+        case Success(updates) =>
+
+          val newLastDate = getLastDate(currentLast, updates)
+          self ! UpdateChatDataLastTime(newLastDate)
+
+          updates.foreach {
+            case ChatData.Update(chat, post, None) =>
+              subscribedBot ! PostReply(chat.id, post)
+            case ChatData.Update(chat, post, Some(prevMessageId)) =>
+              if (updateTgMessages) {
+                subscribedBot ! PostEdit(chat.id, prevMessageId, post)
+              }
+          }
         case Failure(e) => log.error(s"$e")
       }
-      lastUpdateDate = DateUtils.add(lastUpdateDate, subscriptionReplyInterval)
-    case PostsUpdating(posts) => chatData.updatePosts(posts)
+    case UpdateChatDataLastTime(date) =>
+      chatDataLastTime = date
+    case PostsUpdating(posts) =>
+      chatData.updatePosts(posts)
+    case PostWasSentToTg(event) =>
+      chatData.addEvent(event)
   }
+
+  private def getLastDate(default: Date, dates: Seq[ChatData.Update]): Date =
```

Comment:
It is more transparent to have function which returns max of dates in DateUtils and use it here